### ISSUE TITLE
Response from login call back is Error: Invalid SAML Response.

### ISSUE THREAD

**Issue by crwallace** [department-of-veterans-affairs/vets-api#216]
In the login process, the front end is sending the SAMLResponse to `/auth/saml/callback` and I am getting a 403 error and another error stating `Error: Invalid SAML Response. Not match the saml-schema-protocol-2.0.xsd`.

This seemed to be working yesterday morning but now it is not working in any environment.

**Comment by omgitsbillryan** [comment]
We hand the value of that parameter directly into [ruby-saml](https://github.com/onelogin/ruby-saml) and the error message you're seeing `403 - Error: Invalid SAML Response. Not match the saml-schema-protocol-2.0.xsd` gets returned when ruby-saml thinks the SAMLResponse is invalid.

My immediate suspicion is that some kind of url/form encoding/decoding may have altered the value of SAMLResponse before it made its way back to the vets-api

**Comment by omgitsbillryan** [comment]
TL;DR -
There was an issue on the ID.me side and Tanel has fixed it.

Long Version -
The `ID` attribute inside the <samlp:Response> tag has the same value as the `ID` attribute inside the <saml:Assertion> tag and <ds:Reference> tag.  I'm not enough of a SAML spec expert to tell you _why_ that's not allowed, but it isn't, and our ruby-saml library threw a validation error as a result.

**Comment by omgitsbillryan** [comment]
I'm closing.
